### PR TITLE
CSV export

### PR DESCRIPTION
**Pull request by lfilmeyer**
Adds ability to export survey results to CSV
https://github.com/codeforboston/urban-league-heat-pump-accelerator/issues/612 

Sample of exported CSV is something like
```
Survey Visit ID,Home ID,Successful Export,Public Survey,Assignment ID,Assignment Surveyor IDs,Assignment Surveyor Names,Street Number,Street Name,Unit Number,City,State,ZIP Code,Home Latitude,Home Longitude,Survey Visit Latitude,Survey Visit Longitude,Survey Visit Time,Surveyor ID,Surveyor Name,"1. Have you heard of heat pumps?","2. If yes, have you considered them for home heating/cooling?","3. If yes, how far have you taken your interest in heat pumps?","4. Have you heard of Mass Save… the energy efficiency program?","5. If yes, have you had contact with them?","6. If you could save money on your utility bills with a state-discounted or free heating/cooling system upgrade, would you like to learn more?","7. If yes, please confirm your name","8. If yes, please confirm your email address","9. If yes, please confirm your phone number","10. So we can report back on your potential savings, what type of heating system do you have? (Across the whole home, not just in individual rooms)","11. If other, please specify","12. What type of heating FUEL do you use for that heating system?","13. If other, please specify","14. We expect to have “heat pump coaches” available to walk through the process with you (volunteers, who are not affiliated with any installer or brand). Would you like to know more about how to get free, personal help from a heat pump coach?","15. If yes, please confirm your name","16. If yes, please confirm your email address","17. If yes, please confirm your phone number"
1,53,Yes,No,1,"1,2","Alice Abbott,Bob Baker",88,MORELAND ST,,ROXBURY,MA,02119,42.32357123,-71.07763958,42.3729804,-71.1186159,2025-02-07 11:11:15 -0500,1,Alice Abbott,Yes,No,,No,No,No,,,,Radiators – water,,Electric (for baseboard),,No,,,
2,53,Yes,No,1,"1,2","Alice Abbott,Bob Baker",88,MORELAND ST,,ROXBURY,MA,02119,42.32357123,-71.07763958,42.3729804,-71.1186159,2025-02-07 11:11:15 -0500,1,Alice Abbott,Yes,No,,No,No,No,,,,Radiators – water,,Electric (for baseboard),,No,,,
3,52,Yes,No,1,"1,2","Alice Abbott,Bob Baker",96,MORELAND ST,,ROXBURY,MA,02119,42.32327659,-71.07706677,42.3729804,-71.1186159,2025-02-07 11:11:37 -0500,1,Alice Abbott,No,No,,No,No,No,,,,Forced hot air – vents,,Gas,,,,,
4,7435,Yes,Yes,,,,1,Federal St,,Boston,MA,02110,,,,,2025-02-14 13:12:41 -0500,,,Yes,Yes,To the moon,Yes,Yes,Yes,John Smith,john.smith@example.com,1112223333,Other,Fire,Other,Wood,Yes,John Smith,john.smith@example2.com,1142223333
,1,Yes,,1,"1,2","Alice Abbott,Bob Baker",6,WARREN PL,,ROXBURY,MA,02119,42.32811808,-71.08244678,,,,,,,,,,,,,,,,,,,,,,
,2,Yes,,1,"1,2","Alice Abbott,Bob Baker",8,WARREN PL,,ROXBURY,MA,02119,42.3281053,-71.08229235,,,,,,,,,,,,,,,,,,,,,,
,3,Yes,,1,"1,2","Alice Abbott,Bob Baker",14,WARREN PL,,ROXBURY,MA,02119,42.32807088,-71.08184853,,,,,,,,,,,,,,,,,,,,,,
,4,Yes,,1,"1,2","Alice Abbott,Bob Baker",19,WARREN PL,,ROXBURY,MA,02119,42.3277615,-71.08176327,,,,,,,,,,,,,,,,,,,,,,
,5,Yes,,1,"1,2","Alice Abbott,Bob Baker",96 96B,WARREN ST,,ROXBURY,MA,02119,42.32715273,-71.08284705,,,,,,,,,,,,,,,,,,,,,,
```

### CODE REVIEW

**Comment by lfilmeyer** [comment]
Need to test public survey & have a column to flag it in CSV - surveyor id would be null

**Comment by lfilmeyer** [comment]
Nice to have in separate PR - flag certain survey visits that are far from each home - how far is too far? question for Chris

File: backend/app/controllers/admin/exports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Admin
+  class ExportsController < ApplicationController
+    def export
+      @exporter = Exporter.new(survey_id: 1)
+      authorize @exporter
+
+      file_path =  "tmp/storage/export-#{Time.now.strftime("%m-%d-%Y.%H.%M.%S")}.csv"
```
Comment (lfilmeyer) [inline backend/app/controllers/admin/exports_controller.rb:9]: if we do write to disk - maybe use `Tempfile`

Comment (lfilmeyer) [reply backend/app/controllers/admin/exports_controller.rb:9]: do we have to worry about cleaning up tmp directory?

Comment (lfilmeyer) [reply backend/app/controllers/admin/exports_controller.rb:9]: https://ruby-doc.org/3.4.1/stdlibs/tempfile/Tempfile.html

File: backend/.DS_Store
Comment (mzagaja) [inline backend/.DS_Store:1]: might want to add to .gitignore

Comment (lfilmeyer) [reply backend/.DS_Store:1]: i thought i already did 🤔 ... but yeah this was me being sloppy

File: backend/app/controllers/admin/exports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Admin
+  class ExportsController < ApplicationController
+    def export
+      @csv_exporter = CsvExporter.new(survey: Survey.last)
```
Comment (mzagaja) [inline backend/app/controllers/admin/exports_controller.rb:6]: For future maybe take a param or route `/1` and then find the relevant Survey to export with Survey.find.

**Review by mzagaja — APPROVED** [review]
🚢 it!
